packages/msgraph/src/Adaptor.js

```diff
@@ -76,14 +85,195 @@ export function create(resource, data, callback) {
-    const resolveQuery = expandReferences(query)(state);
     const { accessToken, apiVersion } = state.configuration;
+    const [resolvedPath, resolvedQuery] = expandReferences(state, path, query);
+
+    const url = getUrl(resolvedPath, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...resolvedQuery, ...auth }).then(response =>
+      handleResponse(response, state, callback)
+    );
+  };
+}
+
+/**
+ * Get a Drive or SharePoint document library. The drive metadata will be written
+ * to state.drives, where it can be used by other adaptor functions.
+ * Pass { id } to get a drive by id or { id, owner } to get default drive for
+ * some parent resource, like a group
+ * @public
+ * @example <caption>Get a drive by ID</caption>
+ * getDrive({ id: "YXzpkoLwR06bxC8tNdg71m" })
+ * @example <caption>Get the default drive for a site</caption>
+ * getDrive({ id: "openfn.sharepoint.com", owner: "sites" })
+ * @param specifier {Object} - A definition of the drive to retrieve
+ *    - id {string} - The ID of the resource or owner.
+ *    - owner {string} - The type of drive owner (e.g. sites, groups).
+ * @param {string} name - The local name of the drive used to write to state.drives, ie, state.drives[name]
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getDrive(specifier, name = 'default', callback = s => s) {
+  return state => {
+    const { accessToken, apiVersion } = state.configuration;
+    const [resolvedSpecifier, resolvedName] = expandReferences(
+      state,
+      specifier,
+      name
+    );
+
+    const { id, owner = 'drive' } = resolvedSpecifier;
+
+    let urlPath;
+    if (owner === 'drive') {
+      urlPath = `drives/${id}`;
+    } else {
+      urlPath = `${owner}/${id}/drive`;
+    }
+
+    const url = getUrl(urlPath, apiVersion);
+    const auth = getAuth(accessToken);
+
+    return request(url, { ...auth }).then(response => {
+      state.drives[resolvedName] = response;
+      return callback(state);
+    });
+  };
+}
+
+/**
+ * Get folder metadata or an array of folder contents.
+ * @public
+ * @example <caption>Get a folder by ID</caption>
+ * getFolder('01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3')
+ * @example <caption>Get a folder for a named drive by id</caption>
+ * getFolder("01LUM6XOCKDTZKQC7AVZF2VMHE2I3O6OY3",{ driveName: "mydrive"})
+ * @param {string} pathOrId - A path to a folder or folder id
+ * @param {object} options - (Optional) Query parametes
+ * @param {function} [callback = s => s] (Optional) Callback function
+ * @return {Operation}
+ */
+export function getFolder(pathOrId, options, callback = s => s) {
+  return async state => {
+    const defaultOptions = {
+      driveName: 'default', // Named drive in state.drives
+      metadata: false, // If false return folder files if true return folder metadata
+      $filter: '', // Eg: "file/mimeType eq \'application/vnd.ms-excel\'"
```

Comment:
Did we get filter working? I can't see any logic for it. If not we should probably comment this out and add a TODO.

Reply:
We have an issue that track all todo's https://github.com/OpenFn/adaptors/issues/310, So mean while i have commented out `$filter`